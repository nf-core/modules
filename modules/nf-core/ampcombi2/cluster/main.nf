process AMPCOMBI2_CLUSTER {
    tag 'ampcombi2'
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ampcombi:0.2.2--pyhdfd78af_0':
        'biocontainers/ampcombi:0.2.2--pyhdfd78af_0' }"

    input:
    path(summary_file)

    output:
    path("Ampcombi_summary_cluster.tsv")                   , emit: cluster_tsv
    path("Ampcombi_summary_cluster_representative_seq.tsv"), emit: rep_cluster_tsv
    path("Ampcombi_cluster.log")                           , emit: log, optional:true
    path "versions.yml"                                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    ampcombi cluster \\
        --ampcombi_summary ${summary_file} \\
        $args \\
        --threads ${task.cpus}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ampcombi: \$(ampcombi --version | sed 's/ampcombi //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    touch Ampcombi_summary_cluster.tsv
    touch Ampcombi_summary_cluster_representative_seq.tsv
    touch Ampcombi_cluster.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ampcombi: \$(ampcombi --version | sed 's/ampcombi //')
    END_VERSIONS
    """
}
