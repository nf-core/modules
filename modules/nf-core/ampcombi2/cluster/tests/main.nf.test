nextflow_process {

    name "Test Process AMPCOMBI2_CLUSTER"
    script "../main.nf"
    process "AMPCOMBI2_CLUSTER"

    tag "modules"
    tag "modules_nfcore"
    tag "ampcombi2"
    tag "ampcombi2/cluster"
    tag "ampcombi2/complete"

    setup {
        run("AMPCOMBI2_COMPLETE") {
            script "../../../ampcombi2/complete/main.nf"
            process {
                """
                input[0] =
                    [
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombi2/sample_1_ampcombi.tsv', checkIfExists: true),
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombi2/sample_2_ampcombi.tsv', checkIfExists: true)
                    ]
                """
            }
        }
    }

    test("ampcombi2_cluster - metagenome") {
        when {
            process {
                """
                input[0] = AMPCOMBI2_COMPLETE.out.tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.cluster_tsv[0]).readLines()[0].contains("Linear/Cyclic/Branched"),
                    file(process.out.rep_cluster_tsv[0]).readLines()[0].contains("total_cluster_members"),
                    process.out.versions).match() }
            )
        }
    }

    test("ampcombi2_cluster - metagenome - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = AMPCOMBI2_COMPLETE.out.tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
