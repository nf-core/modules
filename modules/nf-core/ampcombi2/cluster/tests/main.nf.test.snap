{
    "ampcombi2_cluster - metagenome": {
        "content": [
            true,
            true,
            [
                "versions.yml:md5,4e9aa3812bfee6ec22a1b6ccb62de2ca"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T12:11:40.928513749"
    },
    "ampcombi2_cluster - metagenome - stub": {
        "content": [
            {
                "0": [
                    "Ampcombi_summary_cluster.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "Ampcombi_summary_cluster_representative_seq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "Ampcombi_cluster.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [
                    "versions.yml:md5,4e9aa3812bfee6ec22a1b6ccb62de2ca"
                ],
                "cluster_tsv": [
                    "Ampcombi_summary_cluster.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log": [
                    "Ampcombi_cluster.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "rep_cluster_tsv": [
                    "Ampcombi_summary_cluster_representative_seq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,4e9aa3812bfee6ec22a1b6ccb62de2ca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T12:12:08.780718892"
    }
}